### PR TITLE
LPD-27976 Date fields must be the ISO8601 pattern in the actions payload

### PR DESCRIPTION
**Pull request by marcelabc**
Related issue: https://liferay.atlassian.net/browse/LPD-27976

### CODE REVIEW

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/dto/v1_0/converter/ObjectEntryDTOConverter.java [hunk 1]
```diff
@@ -795,7 +795,9 @@ private Map<String, Object> _toProperties(
 				map.put(objectFieldName, fileEntry);
 			}
 			else if (objectField.compareBusinessType(
-						ObjectFieldConstants.BUSINESS_TYPE_DATE_TIME)) {
+						ObjectFieldConstants.BUSINESS_TYPE_DATE) ||
+					 objectField.compareBusinessType(
+						 ObjectFieldConstants.BUSINESS_TYPE_DATE_TIME)) {
```
Comment (carolmariaabb) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/dto/v1_0/converter/ObjectEntryDTOConverter.java:800]: @marcelabc are we causing a breaking change by changing the headless API return?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java [hunk 2]
```diff
@@ -2320,6 +2331,12 @@ private void _assertWebhookObjectAction(
 			status,
 			JSONUtil.getValue(
 				payloadJSONObject, "JSONObject/objectEntry", "Object/status"));
+		Assert.assertEquals(
+			birthday,
+			JSONUtil.getValue(
+				payloadJSONObject,
+				"JSONObject/objectEntryDTO" + objectDefinition.getShortName(),
+				"JSONObject/properties", "Object/birthday"));
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java:2339]: I don´t understand the order that already existed here, so it looks good to me

image

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:report:12500335

**Comment by marcelabc** [comment]
I need to check some failures.

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
@marcelabc failures that can be related:

1. [LocalFile.ObjectPortlet#CanViewDateEntryOnLayout](https://testray.liferay.com/#/project/34754/routines/34756/build/12844097/case-result/12855529/history)
2. [LocalFile.ObjectCustomLayouts#CanViewDateEntry](https://testray.liferay.com/#/project/34754/routines/34756/build/12844097/case-result/12855167/history)

**Comment by marcelabc** [comment]
@carolmariaabb, thank you!
The failures are related.

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:object

**Comment by marcelabc** [comment]
ci:report:21377132

**Comment by marcelabc** [comment]
ci:forward

**Pull request closed without merging**